### PR TITLE
Catalog UCS rackmount server's children

### PR DESCRIPTION
**Pull request by tannoa2**
-Fix the URL for the Catalog.
-Catalog the server and its children instead of the server alone.

### CODE REVIEW

**Comment by tannoa2** [comment]
@RackHD/corecommitters @RackHD/veyron

File: lib/jobs/ucs-catalog.js [hunk 1]
```diff
@@ -80,19 +80,18 @@ function UcsCatalogJobFactory(
...
-                var url= baseurl + "/catalog/?host=" + ucs+ "&user="+ username + "&password=" + password + "&identifier="+ node.name;/* jshint ignore:line */
+                var url=  "/catalog?host=" + ucs+ "&user="+ username + "&password=" + password + "&identifier="+ node.name;/* jshint ignore:line */
                 return self.ucs.clientRequest(url);
             })
             .then(function(res) {
                 assert.object(res);
                 return res.body;
             })
-            .then(function(data) {
+            .map(function(data) {
```
Comment (BillyAbildgaard) [inline lib/jobs/ucs-catalog.js]: isn't the `.map` function only for arrays?  Looks like you then assert data should be an object so I'm not sure if that makes sense

Comment (tannoa2) [inline lib/jobs/ucs-catalog.js]: Hi @BillyAbildgaard, correct. We changed the res.body content into an array from the UCS server side, that why we are using the .map

File: lib/jobs/ucs-catalog.js [hunk 2]
```diff
@@ -80,19 +80,18 @@ function UcsCatalogJobFactory(
                 var username = self.ucs.settings.username;
                 var password = self.ucs.settings.password;
                 var ucs = self.ucs.settings.ucs;
-                var baseurl= self.ucs.settings.protocol+"://"+self.ucs.settings.host+ ":"+self.ucs.settings.port;/* jshint ignore:line */
-                var url= baseurl + "/catalog/?host=" + ucs+ "&user="+ username + "&password=" + password + "&identifier="+ node.name;/* jshint ignore:line */
+                var url=  "/catalog?host=" + ucs+ "&user="+ username + "&password=" + password + "&identifier="+ node.name;/* jshint ignore:line */
```
Comment (geoff-reid) [inline lib/jobs/ucs-catalog.js]: Why mark this line as jshint ignore?

Comment (tannoa2) [inline lib/jobs/ucs-catalog.js]: Too Long for hound's taste

Comment (tannoa2) [inline lib/jobs/ucs-catalog.js]: Too Long for hound's taste

**Pull request merged**